scripts/issue/474/upgrade_arb_crv_univ3.py

```diff
@@ -0,0 +1,128 @@
+    proxy_admin = interface.IProxyAdmin(PROXY_ADMIN, owner=safe.account)
+    strat_proxy = interface.ICrvStrategy(strategy_proxy_address, owner=safe.account)
+    controller_proxy = interface.IController(strat_proxy.controller(), owner=vault)
+
+    prev_strategist = strat_proxy.strategist()
+    prev_controller = strat_proxy.controller()
+    prev_gov = strat_proxy.governance()
+    prev_guardian = strat_proxy.guardian()
+    prev_keeper = strat_proxy.keeper()
+    prev_perFeeG = strat_proxy.performanceFeeGovernance()
+    prev_perFeeS = strat_proxy.performanceFeeStrategist()
+    prev_reward = strat_proxy.reward()
+    prev_unit = strat_proxy.uniswap()
+    prev_gauge = strat_proxy.gauge()
+    prev_gauge_factory = strat_proxy.gaugeFactory()
+    prev_swapr_router = strat_proxy.SWAPR_ROUTER()
+
+    if simulation:
+        # Harvest to clear any pending rewards for fresh test case
+        gauge = strat_proxy.gauge()
+        strat_proxy.harvest()
+
+        # Harvest on old strat, store gain in want
+        want = interface.IERC20(gauge)
+        prev_want_bal = want.balanceOf(strat_proxy.address)
+
+        chain.sleep(60*60*2)
+        chain.mine()
+
+        strat_proxy.harvest()
+
+        after_want_bal = want.balanceOf(strat_proxy.address)
+        old_path_want_gain = after_want_bal - prev_want_bal
+
+        # Withdraw want change to keep same conditions for future test
+        controller_proxy.withdraw(vault.token(), old_path_want_gain)
+        assert want.balanceOf(strat_proxy.address) == prev_want_bal
+
+    proxy_admin.upgrade(strategy_proxy_address, logic_address)
+
+    strat_proxy.setUniV3Allowance()
+
+    assert prev_strategist == strat_proxy.strategist()
+    assert prev_controller == strat_proxy.controller()
+    assert prev_gov == strat_proxy.governance()
+    assert prev_guardian == strat_proxy.guardian()
+    assert prev_keeper == strat_proxy.keeper()
+    assert prev_perFeeG == strat_proxy.performanceFeeGovernance()
+    assert prev_perFeeS == strat_proxy.performanceFeeStrategist()
+    assert prev_reward == strat_proxy.reward()
+    assert prev_unit == strat_proxy.uniswap()
+    assert prev_swapr_router == strat_proxy.SWAPR_ROUTER()
+    assert prev_gauge == strat_proxy.gauge()
+    assert prev_gauge_factory == strat_proxy.gaugeFactory()
+    assert UNIV3_ROUTER == strat_proxy.UNIV3_ROUTER()
+
+    if simulation:
+        # Harvest on new strat, store gain in want, compare to prev swap (should be more efficient)
+        prev_want_bal = want.balanceOf(strat_proxy.address)
+
+        chain.sleep(60*60*2)
+        chain.mine()
+        
+        strat_proxy.harvest()
+
+        after_want_bal = want.balanceOf(strat_proxy.address)
+        new_path_want_gain = after_want_bal - prev_want_bal
+
+        console.print(f"Old want gain: {old_path_want_gain}")
+        console.print(f"New want gain: {new_path_want_gain}")
+
+        # Compare
+        assert new_path_want_gain >= old_path_want_gain
```

Comment:
```
Old want gain: 2408928194202095
New want gain: 2397514317489925
  File "brownie/_cli/run.py", line 50, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/issue/474/upgrade_arb_crv_univ3.py", line 30, in main
    upgrade_strategy_logic(
  File "./scripts/issue/474/upgrade_arb_crv_univ3.py", line 126, in upgrade_strategy_logic
    assert new_path_want_gain >= old_path_want_gain
AssertionError: 
```

Reply:
seems this is fixed by letting the chain sleep longer; `60 * 60 * 20` did the trick for me.